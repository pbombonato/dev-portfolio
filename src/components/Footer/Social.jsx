import React from "react";
import { GitHub, LinkedIn, Mail } from '@mui/icons-material'

import SocialIcon from "./SocialIcon";

function Social(props) {
    const socialMediaArray = [
        {
            title: 'GitHub',
            link: 'https://www.github.com/pbombonato/',
            icon: <GitHub style={{fontSize: '2.2rem'}} />
        },
        {
            title: 'LinkedIn',
            link: 'https://www.linkedin.com/in/pablobombonato',
            icon: <LinkedIn style={{fontSize: '2.2rem'}} />
        },
        {
            title: 'Mail',
            link: 'mailto:pablo.h.bombonato@gmail.com',
            icon: <Mail style={{fontSize: '2.2rem'}} />
        }
    ]

    return (
        <div className="row py-5">
            <div className="col-12 d-flex justify-content-center">
                {socialMediaArray.map(e => { return (
                    <SocialIcon {...e} />
                )})}
            </div>
        </div>
    )
}

export default Social