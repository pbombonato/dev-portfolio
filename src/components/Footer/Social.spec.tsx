import React from "react";
import Social from "./Social";
import { render } from "@testing-library/react";

describe("<Social />", () => {
  it("should match snapshot", () => {
    const { container } = render(<Social />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
