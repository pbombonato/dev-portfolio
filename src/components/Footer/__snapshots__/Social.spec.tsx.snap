// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Social /> should match snapshot 1`] = `
<div
  class="row py-5"
>
  <div
    class="col-12 d-flex justify-content-center"
  >
    <a
      class="px-2"
      href="https://www.github.com/pbombonato/"
      rel="noopener"
      target="_blank"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
        style="border: 2px solid #fff; padding: 1.5rem; width: 2.2rem; height: 2.2rem; color: rgb(255, 255, 255);"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="GitHubIcon"
          focusable="false"
          style="font-size: 2.2rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
          />
        </svg>
      </div>
    </a>
    <a
      class="px-2"
      href="https://www.linkedin.com/in/pablobombonato"
      rel="noopener"
      target="_blank"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
        style="border: 2px solid #fff; padding: 1.5rem; width: 2.2rem; height: 2.2rem; color: rgb(255, 255, 255);"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LinkedInIcon"
          focusable="false"
          style="font-size: 2.2rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
          />
        </svg>
      </div>
    </a>
    <a
      class="px-2"
      href="mailto:pablo.h.bombonato@gmail.com"
      rel="noopener"
      target="_blank"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
        style="border: 2px solid #fff; padding: 1.5rem; width: 2.2rem; height: 2.2rem; color: rgb(255, 255, 255);"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MailIcon"
          focusable="false"
          style="font-size: 2.2rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
          />
        </svg>
      </div>
    </a>
    <a
      class="px-2"
      href="https://wa.me/5516988289579"
      rel="noopener"
      target="_blank"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
        style="border: 2px solid #fff; padding: 1.5rem; width: 2.2rem; height: 2.2rem; color: rgb(255, 255, 255);"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="WhatsAppIcon"
          focusable="false"
          style="font-size: 2.2rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.75 13.96c.25.13.41.2.46.3.06.11.04.61-.21 1.18-.2.56-1.24 1.1-1.7 1.12-.46.02-.47.36-2.96-.73-2.49-1.09-3.99-3.75-4.11-3.92-.12-.17-.96-1.38-.92-2.61.05-1.22.69-1.8.95-2.04.24-.26.51-.29.68-.26h.47c.15 0 .36-.06.55.45l.69 1.87c.06.13.1.28.01.44l-.27.41-.39.42c-.12.12-.26.25-.12.5.12.26.62 1.09 1.32 1.78.91.88 1.71 1.17 1.95 1.3.24.14.39.12.54-.04l.81-.94c.19-.25.35-.19.58-.11l1.67.88M12 2a10 10 0 0 1 10 10 10 10 0 0 1-10 10c-1.97 0-3.8-.57-5.35-1.55L2 22l1.55-4.65A9.969 9.969 0 0 1 2 12 10 10 0 0 1 12 2m0 2a8 8 0 0 0-8 8c0 1.72.54 3.31 1.46 4.61L4.5 19.5l2.89-.96A7.95 7.95 0 0 0 12 20a8 8 0 0 0 8-8 8 8 0 0 0-8-8z"
          />
        </svg>
      </div>
    </a>
  </div>
</div>
`;
